src/mxcp/runtime/__init__.py

```diff
@@ -2,123 +2,158 @@
-        if not site_config:
+    def get_secret(self, name: str) -> Optional[Dict[str, Any]]:
+        """Get secret parameters by name."""
+        context = get_execution_context()
+        if not context:
+            raise RuntimeError("No execution context available - function not called from MXCP executor")
+
+        site_config = context.get("site_config")
+        user_config = context.get("user_config")
+        
+        if not user_config or not site_config:
             return None
-            
+        
+        # Get project and profile from site config
         project = site_config.get("project")
         profile = site_config.get("profile")
         
+        if not project or not profile:
+            return None
+        
         try:
+            # Navigate to the secrets in user config: projects -> project -> profiles -> profile -> secrets
             project_config = user_config["projects"][project]
             profile_config = project_config["profiles"][profile]
             secrets = profile_config.get("secrets", [])
             
-            # Secrets is now an array of secret objects
+            # Find secret by name and return its parameters
             for secret in secrets:
-                if secret.get("name") == key:
-                    # Return the entire parameters dict for any secret type
+                if secret.get("name") == name:
                     return secret.get("parameters", {})
-                    
+            
             return None
         except (KeyError, TypeError):
             return None
     
     def get_setting(self, key: str, default: Any = None) -> Any:
-        """Get a configuration setting from site config"""
-        site_config = _site_config_context.get()
+        """Get configuration setting from site config."""
+        context = get_execution_context()
+        if not context:
+            raise RuntimeError("No execution context available - function not called from MXCP executor")
+        
+        site_config = context.get("site_config")
         if not site_config:
             return default
-        return site_config.get(key, default)
+        
+        # Support nested key access (e.g., "dbt.enabled")
+        if "." in key:
+            keys = key.split(".")
+            value = site_config
+            for k in keys:
+                if isinstance(value, dict) and k in value:
+                    value = value[k]
+                else:
+                    return default
+            return value
+        else:
+            return site_config.get(key, default)
     
     @property
-    def user_config(self) -> Optional['UserConfig']:
-        """Get the full user configuration"""
-        return _user_config_context.get()
+    def user_config(self) -> Optional[Dict[str, Any]]:
+        """Access full user configuration."""
+        context = get_execution_context()
+        if not context:
+            return None
```

Comment:
That's the property one would use from an `@on_init` hook right?

Why not failing with "function not called from MXCP executor" (like in `get_secret`)?